src/closed.jl

```diff
@@ -0,0 +1,51 @@
+"""
+A `ClosedInterval(left, right)` is an interval set that includes both its upper and lower bounds. In 
+mathematical notation, the constructed range is `[left, right]`.
+"""
+immutable ClosedInterval{T}
+    left::T
+    right::T
+end
+
+ClosedInterval(left, right) = ClosedInterval(promote(left, right)...)
+
+..(x, y) = ClosedInterval(x, y)
+
+±(x, y) = ClosedInterval(x - y, x + y)
+±(x::CartesianIndex, y) = map(ClosedInterval, (x - y).I, (x + y).I)
+
+show(io::IO, I::ClosedInterval) = print(io, I.left, "..", I.right)
+
+in(v, I::ClosedInterval) = I.left <= v <= I.right
+
+isempty(A::ClosedInterval) = A.left > A.right
+
+isequal(A::ClosedInterval, B::ClosedInterval) = (isequal(A.left, B.left) & isequal(A.right, B.right)) | isempty(A) & isempty(B)
+
+==(A::ClosedInterval, B::ClosedInterval) = (A.left == B.left && A.right == B.right) || isempty(A) && isempty(B)
```

Comment:
Just checking, does the operator precedence work out? Might want to add a test `I == I` and/or `L == L` below.